python/lsst/pipe/drivers/singleFrameDriver.py

```diff
@@ -14,11 +14,13 @@ class SingleFrameDriverTask(BatchParallelTask):
     """
     ConfigClass = SingleFrameDriverConfig
     _DefaultName = "singleFrameDriver"
+    RunnerClass = ButlerInitializedTaskRunner
 
     def __init__(self, *args, **kwargs):
+        butler = kwargs.pop("butler")
```

Comment:
Good point.  Given that I believe processCcd doesn't actually require a butler in some configurations, maybe we should just make it optional by giving it a None default:
```
butler = kwargs.pop("butler", None)
```
On a somewhat related note, I've just noticed the new requirement for a butler argument to `CharacterizeImageTask.__init__`, which I think should also be optional.  Perhaps we should do the same thing there (and maybe `ProcessCcdTask` and `CalibrateTask`)?